### ISSUE TITLE
Docs for uniform_ don't make any sense

### ISSUE THREAD

**Issue by ezyang** [pytorch/pytorch#33898]
It says `P(x) = 1/(to - from)` but this formula is only true for discrete types (also there needs to be a condition on x anyway)

cc @fritzo @neerajprad @alicanb @vishwakftw

**Comment by SsnL** [comment]
Density is also `1/(to-from)` for continuous dtypes.

**Comment by vincentqb** [comment]
I don't seem to find the place you are referring to, [here](https://github.com/pytorch/pytorch/blob/master/torch/distributions/uniform.py) or [here](https://github.com/pytorch/pytorch/blob/master/torch/nn/init.py#L74) or elsewhere?

**Comment by ezyang** [comment]
It's in `torch/_tensor_docs.py` 

```

add_docstr_all('uniform_',
               r"""
uniform_(from=0, to=1) -> Tensor

Fills :attr:`self` tensor with numbers sampled from the continuous uniform
distribution:

.. math::
    P(x) = \dfrac{1}{\text{to} - \text{from}}
""")

```

**Comment by vincentqb** [comment]
Can you mention what you would expect? `P(x) = 1/(to - from)` seems correct to me. Since picking x between `[from, to]` uniformly at random means that each `x` must have the same probability `p`. Since the density `P(x)` must sum to 1, `p*(to - from)=1` so `p=1/(to-from)`. Is that what you meant?

**Comment by ezyang** [comment]
I'm wrong! Sorry about the false alarm.

**Comment by vincentqb** [comment]
Glad we clarified :)

**Comment by SsnL** [comment]
Looking at this again, I think we should add an indicator function. The density is only that value when `x` is between ` to ` and `from`.

**Comment by vincentqb** [comment]
Indeed, nothing prevents the user from evaluating `P(x)` outside of `[to, from]`.
